server/controllers/v1/user.ts

```diff
@@ -9,13 +11,117 @@ export enum TOKEN_TYPE {
 
 export default class UserController {
   protected userRepo: typeof User;
+  secret: string;
   protected emailService: EmailService;
 
   constructor(model: typeof User, emailService: EmailService) {
     this.userRepo = model;
+    this.secret = process.env.SECRET ? process.env.SECRET : "secret";
```

Comment:
Not secure to use a default secret - we should error out if one hasn't been provided. You could image a deploy procedure which omits the secret and therefore it always defaults to "secret". In cases where the ENV variable is set, but not passed in would hide this issue - you may think it's being passed in, but it's not. Further, if this is the case, when you rotate the secret, it never is actually rotated since it's always the string "secret"

Reply:
Fixed, an error should now be thrown when secret is undefined.